src/Aspire.Hosting.Azure/AzurePublishingContext.cs

```diff
@@ -217,9 +217,17 @@ static BicepValue<string> ResolveValue(object val)
 
         var outputs = new Dictionary<string, BicepOutputReference>();
 
+        void CaptureBicepOutputs(object value)
+        {
+            if (value is BicepOutputReference bo)
+            {
+                outputs[bo.ValueExpression] = bo;
+            }
+        }
+
         foreach (var resource in model.Resources)
         {
-            if (resource.GetDeploymentTargetAnnotation()?.DeploymentTarget is AzureBicepResource br)
+            if (resource.GetDeploymentTargetAnnotation() is { } annotation && annotation.DeploymentTarget is AzureBicepResource br)
```

Comment:
```suggestion
            if (resource.GetDeploymentTargetAnnotation() is { DeploymentTarget: AzureBicepResource br })
```

*insert galaxy-brain gif*

Reply:
![image](https://github.com/user-attachments/assets/63c01a27-6e90-4adc-b46b-4c8136a5afc0)